lib/src/model/result.dart

```diff
@@ -26,8 +25,8 @@ class Result<Value, Failure> {
 
   /// Execute specific actions on success and on failure.
   T iif<T>(
-      {@required T Function(Value value) success,
-      @required T Function(Failure failure) failure}) {
+      {required T Function(Value? value) success,
+      required T Function(Failure? failure) failure}) {
```

Comment:
At least the first line should be reverted. I'm less sure about the second one.

Please, try to understand semantic of this class. Don't just put `?` and `!` until it compiles. If the class requires some more refactoring, it should be done, because this is an important infrastructural class we use a lot in the library.

Reply:
Yep I agree fixed it into: 

```dart
/// Execute specific actions on success and on failure.
  T iif<T>(
      {required T Function(Value value) success,
      required T Function(Failure failure) failure}) {
    assert(_value == null || _failure == null);

    if (_failure != null) {
      return failure(_failure!);
    } else if (_value != null) {
      return success(_value!);
    } else {
      throw Exception('Both value and failure cannot be null');
    }
  }
```